src/main/java/com/manardenza/entity/Reservation.java

```diff
@@ -3,14 +3,31 @@
 import lombok.AllArgsConstructor;
 import lombok.Data;
 
-import java.io.Serializable;
 import java.util.Date;
 
 @Data
 @AllArgsConstructor
-public class Reservation implements Serializable {
+public class Reservation extends AbstractObject {
 
     private Date reservedFrom;
     private Date reservedTo;
     private User reservedUser;
+    private Room reservedRoom;
+    private Hotel reservedHotel;
+
+    public boolean overlaps(Date newReservedFrom, Date newReservedTo) {
```

Comment:
This implementation is error prone. You may use YodaTime Interval to check this, or just simplify the logic. Two ranges overlap if:
(StartDate1 <= EndDate2) && (StartDate2 <= EndDate1)